### ISSUE TITLE
test: Intermittent failure in `feature_index_prune.py`

### ISSUE THREAD

**Issue by aureleoules** [bitcoin/bitcoin#26330]
https://cirrus-ci.com/task/5021369209978880
https://cirrus-ci.com/task/6533221721571328
https://cirrus-ci.com/task/5355957631320064

**Comment by MarcoFalke** [comment]
The bug should be fixed, but this is likely caused by https://github.com/bitcoin/bitcoin/pull/26297#issuecomment-1276174908

**Comment by hebasto** [comment]
Errors are quite different:

- > https://cirrus-ci.com/task/5021369209978880 and https://cirrus-ci.com/task/6533221721571328

```
test_framework.authproxy.JSONRPCException: 'getbestblockhash' RPC took longer than 240.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)
```

- > https://cirrus-ci.com/task/5355957631320064

```
AssertionError: [node 1] Expected messages "['limited pruning to height 2489']" does not partially match log:
```

**Comment by brunoerg** [comment]
https://cirrus-ci.com/task/6147124562755584

`AssertionError: [node 1] Expected messages "['limited pruning to height 2489']"`

**Comment by mzumsande** [comment]
> Errors are quite different:

I agree that these are two unrelated issues. I opened #26417 for the missing `limited pruning to height 2489` log messages, not sure if the timeout issue still needs a fix specific to this test after #26297?
